### ISSUE TITLE
Question about model.train() and model.eval() in Pytorch Lightning

### ISSUE THREAD

**Issue by GillianGrayson** [facebookresearch/nbm-spam#14]
Hi!

I recently started learning Pytorch Lightning in the context of tabular data.
May I ask you a question for understanding?

Is it really necessary to translate the model into train() state here:
https://github.com/facebookresearch/nbm-spam/blob/15d70075ce23d91904b34efffa837317b78bd85e/nbm_spam/train_tabular.py#L297
and in eval() state here:
https://github.com/facebookresearch/nbm-spam/blob/15d70075ce23d91904b34efffa837317b78bd85e/nbm_spam/train_tabular.py#L305
?

The documentation for Pytorch Lightning says that it automatically puts the model into the desired state.
Please clarify whether explicit calls to train() and eval() are really needed here due to the non-trivial implementation of models?

**Comment by filipradenovic** [comment]
Indeed you are right, Pytorch Lightening will do `.eval()` before validation loop and `.train()` before training loop, under the hood. So we could easily remove these two lines. However, we do believe that this makes the code more clear and easier to follow. Eg, if one would use our model to evaluate on their task outside of pytorch ligthening, in a simple notebook, then this line `self.model.eval()` is a good reminder to put model in the eval mode.

**Comment by abhimanyudubey** [comment]
Closing as comment is resolved.
